File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
class Solution(object):
    def traverse(self, root, level, pos, result):
        if not root:
            return

        if len(result) < level + 1:
            result.append([])
        
        result[level].append(root.val)

        self.traverse(root.left, level + 1, pos * 2, result)
        self.traverse(root.right, level + 1, pos * 2 + 1, result)

    def levelOrder(self, root):
        if not root:
            return []

        result = []

        self.traverse(root, 0, 0, result)

        return result

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
class Solution {
public:
    vector<vector<int>> res;
    
    vector<vector<int>> levelOrder(TreeNode* root) {
        check(root, 0);
        return res;
    }

    void check(TreeNode* root, int level) {
        if (!root) return;

        if (res.size() <= level) res.push_back(vector<int>());
        res[level].push_back(root->val);

        check(root->left, level + 1);
        check(root->right, level + 1);
    }
};
